src/services/event.ts

```diff
@@ -105,6 +117,70 @@ async function findEventHelper(where: WhereOptions, plural = false): Promise<Eve
   };
 }
 
+interface FindEventHelperBlobOptions {
+  key: string;
+  value: string;
+  regions: number[] | undefined;
+  fallbackValue?: string;
+  allowNull?: boolean;
+}
+
+async function findEventHelperBlob({
+  key,
+  value,
+  regions,
+  fallbackValue,
+  allowNull = false,
+}: FindEventHelperBlobOptions): Promise<EventShape[]> {
+  const getClause = () => {
+    if (allowNull) {
+      return {
+        [Op.or]: [
+          { [key]: value },
+          { [key]: { [Op.eq]: null } },
+        ],
+      };
+    }
+
+    return { [key]: value };
+  };
+
+  const where = { data: { ...getClause() } };
+
+  if (regions && regions.length) {
+    // @ts-ignore
+    where.regionId = regions;
+  }
+
+  const events = await EventReportPilot.findAll({
+    attributes: [
+      'id',
+      'ownerId',
+      'pocId',
+      'collaboratorIds',
+      'regionId',
+      'data',
+    ],
+    raw: true,
+    where,
```

Comment:
Probably want to do the ORDER BY here for what they want:
Sort order - (first) Event start date, (secondary) event title, sorted oldest